gsa/align.go

```diff
@@ -15,6 +15,31 @@ package gsa
 //      The two rows in the pairwise alignment
 func Align(p, q, edits string) (pRow, qRow string) {
 	pRow, qRow = "", ""
+	pItr, qItr := 0, 0
+	for pos, char := range edits {
+		char := string(char)
+
+		pEnd := pItr + 1
+		qEnd := qItr + 1
+		if char == "M" {
+			pRow += p[pItr:pEnd]
+			qRow += q[qItr:qEnd]
```

Comment:
String concatenation will give you a quadratic time running time (unless they have recently optimised that to work like `append`, in which case I am wrong about that). If I'm right, you don't want to do this, but since Go 1.10 the `strings` module has had a `Builder` type you can use to construct a string iteratively.

Even faster (if you don't mind wasting a tiny amount of memory) is to just allocate a `byte` buffer and put the results there. You know that the buffers you need cannot be longer than `edits`--it is an upper bound on how long the individual strings can be--so `make([]byte, len(edits))` will give you such a buffer.

With the `Builder` you can write `buf.WriteByte(p[pItr])`, and with a buffer you can write `buf[i] = p[pItr]` where `i` is the current length of the string in `buf`.